src/sysjson.go

```diff
@@ -39,9 +50,13 @@ func main() {
 
 func statsHandler(w http.ResponseWriter, r *http.Request) {
 	resp := map[string]interface{}{}
-
-	loadModules(resp, r.URL.Query().Get("modules"))
-
+	
+	if len(*from) == 0 || (len(*from) > 0 && strings.HasPrefix(r.RemoteAddr, *from)) {
+		loadModules(resp, r.URL.Query().Get("modules"))
+	} else {
+		resp["error"] = "Not authorised"
```

Comment:
To follow the pattern that we've already begun to use, I think this should return `http.Error` instead of an error object via JSON.